personal_budget_main.py

```diff
@@ -0,0 +1,214 @@
+            if file_name.endswith(".xls") or file_name.endswith(".xlsx"):
+                excel_file_exist = True
+                file_url = dir_path + "/" + file_name
+
+                # Importing Excel data.
+                warnings.filterwarnings("ignore", ".*DrawingML*.")
+                try:
+                    wb = openpyxl.load_workbook(file_url)
+                    ws = wb.active
+                except Exception as e:
+                    print(f"Error: {e}")
+                    sys.exit(1)
+                
+                for row in ws.iter_rows(min_row=2, values_only=True):  # Assuming headers are in the first row
+                    txn_date, txn_detail = row[0], row[2]
+                    debit = row[5] if row[5] is not None else 0
+                    credit = row[6] if row[6] is not None else 0
+                    if isinstance(txn_date, (datetime.datetime, datetime.date)) and \
+                       (isinstance(debit, (int, float)) or isinstance(credit, (int, float))):
+                        
+                        txn_key = "".join(random.choices(string.ascii_uppercase + string.digits, k=6)) # Unique key to identify individual transaction.
+                        transactions_all[txn_key] = {   "txn_date" : txn_date.strftime("%d-%b-%Y"),
+                                                        "txn_detail" : txn_detail,
+                                                        "debit" : debit,
+                                                        "credit" : credit
+                                                    }
+                        
+                        # Consolidating transactions month-wise.
+                        month_key = txn_date.strftime("%m-%Y")
+                        if month_key not in monthly_summary:
+                            monthly_summary[month_key] = {  "month_name" : txn_date.strftime("%B, %Y"),
+                                                            "txn_keys" : [txn_key],
+                                                            "debit" : debit,
+                                                            "credit" : credit,
+                                                            "surplus_deficit" : credit - debit
+                                                        }
+                        else:
+                            monthly_summary[month_key]["txn_keys"].append(txn_key)
+                            monthly_summary[month_key]["debit"] += debit
+                            monthly_summary[month_key]["credit"] += credit
+                            monthly_summary[month_key]["surplus_deficit"] += credit - debit
+        
+        if not excel_file_exist:
+            sys.exit("No Excel file in source directory")
+    
+    except Exception as e:
+        print(f"Error: {e}")
+        sys.exit(1)
+    
+    return transactions_all, monthly_summary
+
+
+def generate_report(transactions_all, monthly_summary, report_meta, sort_code):
```

Comment:
Break this into 3 functions. if, elif and else blocks as separate functions. Also add function docstrings to each of these 3 functions to explain what each is doing.